client/compliance_client.go

```diff
@@ -585,6 +654,894 @@ func (c *complianceGRPCClient) CancelOperation(ctx context.Context, req *longrun
+}
+
+// RepeatDataQuery this method echoes the ComplianceData request. This method exercises
+// sending all request fields as query parameters.
+func (c *complianceRESTClient) RepeatDataQuery(ctx context.Context, req *genprotopb.RepeatRequest, opts ...gax.CallOption) (*genprotopb.RepeatResponse, error) {
+	m := protojson.MarshalOptions{AllowPartial: true, EmitUnpopulated: true}
+	jsonReq, err := m.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	baseUrl, _ := url.Parse(c.endpoint)
+	baseUrl.Path += fmt.Sprintf("/v1beta1/repeat:query")
+
+	params := url.Values{}
+	if req.GetInfo().GetFBool() {
+		params.Add("info.fBool", fmt.Sprintf("%v", req.GetInfo().GetFBool()))
+	}
+	if req.GetInfo().GetFBytes() != nil {
+		params.Add("info.fBytes", fmt.Sprintf("%v", req.GetInfo().GetFBytes()))
+	}
+	if req.GetInfo().GetFChild().GetFBool() {
+		params.Add("info.fChild.fBool", fmt.Sprintf("%v", req.GetInfo().GetFChild().GetFBool()))
+	}
+	if req.GetInfo().GetFChild().GetFChild().GetFBool() {
+		params.Add("info.fChild.fChild.fBool", fmt.Sprintf("%v", req.GetInfo().GetFChild().GetFChild().GetFBool()))
+	}
+	if req.GetInfo().GetFChild().GetFChild().GetFDouble() != 0 {
+		params.Add("info.fChild.fChild.fDouble", fmt.Sprintf("%v", req.GetInfo().GetFChild().GetFChild().GetFDouble()))
+	}
+	if req.GetInfo().GetFChild().GetFChild().GetFString() != "" {
+		params.Add("info.fChild.fChild.fString", fmt.Sprintf("%v", req.GetInfo().GetFChild().GetFChild().GetFString()))
+	}
+	if req.GetInfo().GetFChild().GetFContinent() != 0 {
+		params.Add("info.fChild.fContinent", fmt.Sprintf("%v", req.GetInfo().GetFChild().GetFContinent()))
+	}
+	if req.GetInfo().GetFChild().GetFDouble() != 0 {
+		params.Add("info.fChild.fDouble", fmt.Sprintf("%v", req.GetInfo().GetFChild().GetFDouble()))
+	}
+	if req.GetInfo().GetFChild().GetFFloat() != 0 {
+		params.Add("info.fChild.fFloat", fmt.Sprintf("%v", req.GetInfo().GetFChild().GetFFloat()))
+	}
+	if req.GetInfo().GetFChild().GetFString() != "" {
+		params.Add("info.fChild.fString", fmt.Sprintf("%v", req.GetInfo().GetFChild().GetFString()))
+	}
+	if req.GetInfo().GetFChild().GetPBool() {
+		params.Add("info.fChild.pBool", fmt.Sprintf("%v", req.GetInfo().GetFChild().GetPBool()))
+	}
+	if req.GetInfo().GetFChild().GetPChild().GetFBool() {
+		params.Add("info.fChild.pChild.fBool", fmt.Sprintf("%v", req.GetInfo().GetFChild().GetPChild().GetFBool()))
+	}
+	if req.GetInfo().GetFChild().GetPChild().GetFDouble() != 0 {
+		params.Add("info.fChild.pChild.fDouble", fmt.Sprintf("%v", req.GetInfo().GetFChild().GetPChild().GetFDouble()))
+	}
+	if req.GetInfo().GetFChild().GetPChild().GetFString() != "" {
+		params.Add("info.fChild.pChild.fString", fmt.Sprintf("%v", req.GetInfo().GetFChild().GetPChild().GetFString()))
+	}
+	if req.GetInfo().GetFChild().GetPContinent() != 0 {
+		params.Add("info.fChild.pContinent", fmt.Sprintf("%v", req.GetInfo().GetFChild().GetPContinent()))
+	}
+	if req.GetInfo().GetFChild().GetPDouble() != 0 {
```

Comment:
For the `P*` fields, which are proto3 `optional`, we want to include them if they have been set, even if they've been set to the zero value. So you probably want `....GetFChild().PDouble != nil`, or better yet, calling a method like `....GetFChild().HasPDouble()` if such a method exists.

Reply:
Such methods do not exist, unfortunately. For optional leaf primitive fields I think we are forced to do the double check of nil and and value.